pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -1504,3 +1504,90 @@ func TestCacheInvalidationRace(t *testing.T) {
 		t.Errorf("Predicate should have been called twice. Was called %d times.", callCount)
 	}
 }
+
+// TestCacheInvalidationRace2 tests that cache invalidation is correctly handled
+// when an invalidation event happens during predicate is running.
```

Comment:
grammar nit: "during predicate is running" should be "while a predicate is running"

Reply:
Updated.